sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -135,6 +147,7 @@ abstract class Expression extends TreeNode[Expression] {
   /**
    * Returns true when two expressions will always compute the same result, even if they differ
    * cosmetically (i.e. capitalization of names in attributes may be different).
+   * TODO: how should this deal with nonDeterministic
```

Comment:
Yeah, I guess we could just add `!nonDeteminstic && ...` since nonDeterministic expressions can't equal anything?